Реализация поведения поля ввода (UI) #66

darkelf007 opened the pull request:
Добавлено поведение кнопки удалить и иконки поиска,
добавлен binding,
добавлен debounce и реализован во ViewModel.
Управление состоянием загрузки осуществляется через LiveData. Реализовано поведение Placeholder.

Commit:
Добавлено поведение кнопки удалить и иконки поиска,
добавлен binding,
добавлен debounce и реализован во ViewModel.
Управление состоянием загрузки осуществляется через LiveData.
Реализовано поведение Placeholder.

ypypyxa reviewed (changes requested):
надо что бы пока набираешь текст - фрагмент ждет, как только 2 секунды ни чего не печатываешь - появляется загрузка и начинается поиск (в поиск можешь вставить как раз логгирование)

Commit:
Исправлено поведение дебаунса,
и добавлен параметр для стиля нижней панели(отдельная коррекция UI)

Commit:
Скрытие плейсхолдера во время срабатывания дебаунса

Commit:
Merge branch 'develop' of https://github.com/5ir1us/android-diplom into DP-Task-66

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/vacancy/search/ui/SearchFragment.kt
#	app/src/main/java/ru/practicum/android/diploma/vacancy/search/ui/SearchViewModel.kt

Commit:
Исправлена логика дебаунса,
проверка пустой строки внутри searchRequest и корректная отмена предыдущих вызовов

ypypyxa reviewed (approved):
👌
